src/Paprika/Chain/SnapSync.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Paprika.Crypto;
+using Paprika.Data;
+using Paprika.Merkle;
+
+namespace Paprika.Chain;
+
+public static class SnapSync
+{
+    public const int BoundaryValueSize = Keccak.Size + PreambleLength;
+    private const byte Byte0 = 0;
+    private const byte Byte1 = 0;
+    private const int PreambleLength = 2;
+
+    public static Span<byte> WriteBoundaryValue(in Keccak value, in Span<byte> payload)
+    {
+        payload[0] = Byte0;
+        payload[1] = Byte1;
+        value.Span.CopyTo(payload[PreambleLength..]);
+
+        return payload.Slice(0, BoundaryValueSize);
+    }
+
+    public static bool CanBeBoundaryLeaf(in Node.Leaf leaf)
+    {
+        return leaf.Path.HasOnlyZeroes();
```

Comment:
When adding a boundary node (path-keccak pair), the path needs to filled up to full width with zeros?

Reply:
Nope. It's takes care of by the `SnapSync` part. Just use methods on the raw.